src/index.js

```diff
@@ -1,15 +1,50 @@
+/* eslint-env serviceworker */
+import * as Hasher from 'fr32-sha2-256-trunc254-padded-binary-tree-multihash'
+
+import * as Digest from 'multiformats/hashes/digest'
+import { Piece } from '@web3-storage/data-segment'
+
 /**
- * Welcome to Cloudflare Workers! This is your first worker.
- *
- * - Run `npm run dev` in your terminal to start a development server
- * - Open a browser tab at http://localhost:8787/ to see your worker in action
- * - Run `npm run deploy` to publish your worker
- *
- * Learn more at https://developers.cloudflare.com/workers/
+ * @typedef {import('./bindings.js').Environment} Environment
+ * @typedef {import('./bindings.js').Context} Context
  */
 
 export default {
+  /**
+   * @param {Request} request
+   * @param {Environment} env
+   * @param {Context} ctx
+   */
   async fetch (request, env, ctx) {
-    return new Response('Hello World!')
+    const hasher = Hasher.create()
+    console.log('hasher', typeof hasher)
+
+    // TODO: UCANTIFY
+    const reqUrl = new URL(request.url)
+    const pathname = reqUrl.pathname
+
+    const cid = pathname.replace('/', '')
```

Comment:
Can we do CID parsing here and checking that it's actually a CAR cid ? Assumption that we have CAR cid after `/` seems error prone.

Reply:
this was mostly for testing. There is still some way to go here